hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -295,7 +318,8 @@ protected void removeAllInternal(Predicate predicate) {
     }
 
     @Override
-    protected void deleteInternal(Data key) {
+    protected void deleteInternal(Object key) {
+        key = serializeKeys ? toDataWithStrategy(key) : key;
```

Comment:
`toDataWithStrategy()` is part of the `MapProxySupport` class and also used there, so we cannot have the `serializeKeys ? toDataWithStrategy(key) : key` part in there. But we could have a new method like this in the `NearCachedMapProxy`:
```java
    private Object toNearCacheKeyWithStrategy(Object key) {
        return serializeKeys ? serializationService.toData(key, partitionStrategy) : key;
    }
```
(I would inline the `toDataWithStrategy()` here, so there is less analysis work for the JIT)